Code/Framework/AzCore/Tests/AZStd/String.cpp

```diff
@@ -1779,7 +1780,7 @@ namespace UnitTest
     {
         using TypeParam = char;
         // null terminated compile time string
-        auto MakeCompileTimeString1 = []() constexpr -> const TypeParam*
+        [[maybe_unused]] auto MakeCompileTimeString1 = []() constexpr -> const TypeParam*
```

Comment:
Well it is used, but only at a compile time, so that is maybe why the clang is flaging it.

Reply:
Actually nevermine, it isn't used in this test.
You can just delete this lambda.

Reply:
yes, since is a test and etc, it should be fine here to leave it marked